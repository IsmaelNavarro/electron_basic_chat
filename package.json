{
  "name": "chat",
  "author": "Ismael Navarro",
  "description": "Simple chat with electron",
  "version": "1.0.0",
  "main": "./app/main.js",
  "build": {
    "appId": "com.example.chat"
  },
  "scripts": {
    "start": "set NODE_ENV=development&& electron .",
    "dev": "webpack -d --watch",
    "build": "webpack",
    "test": "jest",
    "dist": "build"
  },
  "jest": {
    "roots": [
      "<rootDir>/app"
    ]
  },
  "dependencies": {
    "ip": "^1.1.5",
    "nodejs-websocket": "^1.7.1",
    "photonkit": "^0.1.2",
    "prop-types": "^15.5.7",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "react-timeago": "^3.4.3"
  },
  "devDependencies": {
    "babel-core": "^6.22.1",
    "babel-jest": "^21.2.0",
    "babel-loader": "^7.1.2",
    "babel-plugin-transform-class-properties": "^6.23.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-es2017": "^6.22.0",
    "babel-preset-react": "^6.22.0",
    "babel-preset-stage-2": "^6.24.1",
    "css-loader": "^0.28.7",
    "devtron": "^1.4.0",
    "electron": "^1.6.2",
    "electron-builder": "^19.48.3",
    "electron-debug": "^1.1.0",
    "electron-devtools-installer": "^2.2.1",
    "enzyme": "^3.2.0",
    "enzyme-adapter-react-16": "^1.1.0",
    "file-loader": "^1.1.5",
    "jest": "^21.2.1",
    "react-test-renderer": "^16.2.0",
    "style-loader": "^0.19.0",
    "webpack": "^3.10.0"
  }
}
