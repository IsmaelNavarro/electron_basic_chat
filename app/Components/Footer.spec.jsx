import * as React from 'react';
import Footer from './Footer';
import TestRenderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as manifest from '../../package.json';

configure({ adapter: new Adapter() });

describe('Footer', () => {
    it('matches previously saved snapshot', () => {
        const tree = TestRenderer.create(<Footer />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('remders manifest name', () => {
        const tree = shallow(<Footer />);
        expect(tree.find('footer').length).toBe(1);
        expect(tree.find('footer').text().indexOf(manifest.name)).not.toBe(-1);
    });
}); 