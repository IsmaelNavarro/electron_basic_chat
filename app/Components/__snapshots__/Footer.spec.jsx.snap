// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches previously saved snapshot 1`] = `
<footer
  className="toolbar toolbar-footer"
>
  <h1
    className="title"
  >
    chat
     v.
    1.0.0
  </h1>
</footer>
`;
